### ISSUE TITLE
t/07-initialize.t hangs with Term-ReadLine-Gnu-1.26

### ISSUE THREAD

**Issue by ppisar** [PadreIDE/Debug-Client#1]
Upgrading Term-ReadLine-Gnu from 1.25 to 1.26, the t/07-initialize.t dead-locks here:

```
$ prove -b -v t/07-initialize.t
t/07-initialize.t ..
1..4
ok 1 - initialize with prams
LOCK
ok 2 - quit with prams
ok 3 - initialize without prams
ok 4 - quit witout prams
ok
```

This is caused by Term-ReadLine-Gnu commit:

```
r481 | hayashi | 2015-01-31 13:07:49 +0100 (So, 31 led 2015) | 2 lines
make handling of iostreams simple (make _rl_store_iostream() return void and remove _rl_fetch_iostream()) [rt.cpan.org #101078]
```

The issue is that the "c" command is sent by run() method to before debugger emits the "DB<1>" prompt. It also possible it is a bug in the debugger because the prompt is never emitted  if the debugger is run from the t/07-initialize.t test while Term-ReadLine-Gnu-1.26 is installed. If run the debugger from another session, the prompt appears and tests pass.

See https://bugzilla.redhat.com/show_bug.cgi?id=1189459 (perl-Debug-Client bug for Fedora Linux distribution) and https://rt.cpan.org/Public/Bug/Display.html?id=101078 (Term-ReadLine-Gnu bug introducing the Term-ReadLine-Gnu change.

**Comment by kevindawson** [comment]
ppisar
thanks for the bug
I will added to my todo list, thanks

**Comment by ppisar** [comment]
I happy to write that fix for this specific issue published on the Term-ReadLine-Gnu side.

I still think some improvements could be done in the synchronization in the Debug-Client tests to eliminate the seldom and race-prone "sleep" commands.

**Comment by al-berger** [comment]
Is there a way to fix this bug? Padre won't install due to it.

**Comment by ppisar** [comment]
On Tue, Mar 31, 2015 at 05:20:36AM -0700, nbdsp wrote:

> Is there a way to fix this bug? Padre won't install due to it.
> 
> You can apply a patch to Term-ReadLine-Gnu
> https://rt.cpan.org/Public/Bug/Display.html?id=101078#txn-1463705.

-- Petr

**Comment by al-berger** [comment]
Thanks! That worked.

**Comment by nishanta454** [comment]
Please let me know about the files in which i have to apply below mentioned patch. I am unable to find files Gnu.pm or Gnu.xs   either. Thanks in advance...

You can apply a patch to Term-ReadLine-Gnu
https://rt.cpan.org/Public/Bug/Display.html?id=101078#txn-1463705.

**Comment by szabgab** [comment]
Term-ReadLine-Gnu is at 1.35 already. I assume this is not an issue any more.

**Comment by dreggs76** [comment]
It's still an issue with Term-ReadLine-Gnu 1.35, so the suggested fixes for 1.25/1.26 are probably not applicable. Padre (for me) won't install :(
